lib/internal/url.js

```diff
@@ -647,6 +647,35 @@ function getObjectFromParams(array) {
   return obj;
 }
 
+// Mainly to mitigate func-name-matching ESLint rule
+function defineIDLClass(proto, classStr, obj) {
```

Comment:
I'm wondering if there is a possibility/necessity to generate these stuff from an IDL file, like what browsers do, if there will be more convergence with Web APIs in Node core (console, crypto, streams, etc.) Not that this is what this PR should deal with though(and this is just to mitigate func-name-matching anyway, not full-on generation), just a thought.

Also the name is a little bit too general, could be something like `defineIDLData` or `defineIDLValues`, since it's about properties that need to be defined with `value`.